Remotion/Data/DomainObjects.UnitTests/Persistence/Configuration/StorageProviderDefinitionTest.cs

```diff
@@ -35,6 +35,34 @@ namespace Remotion.Data.DomainObjects.UnitTests.Persistence.Configuration
   [TestFixture]
   public class StorageProviderDefinitionTest
   {
+
+    public class StorageObjectFactoryFake : IStorageObjectFactory
+    {
+      public StorageProvider CreateStorageProvider (StorageProviderDefinition storageProviderDefinition, IPersistenceExtension persistenceExtension)
+      {
+        throw new NotImplementedException();
+      }
+
+      public StorageProvider CreateReadOnlyStorageProvider (StorageProviderDefinition storageProviderDefinition, IPersistenceExtension persistenceExtension)
+      {
+        throw new NotImplementedException();
+      }
+
+      public IPersistenceModelLoader CreatePersistenceModelLoader (
+          StorageProviderDefinition storageProviderDefinition, IStorageProviderDefinitionFinder storageProviderDefinitionFinder)
+      {
+        throw new NotImplementedException();
+      }
+
+      public IDomainObjectQueryGenerator CreateDomainObjectQueryGenerator (
+          StorageProviderDefinition storageProviderDefinition,
+          IMethodCallTransformerProvider methodCallTransformerProvider,
+          ResultOperatorHandlerRegistry resultOperatorHandlerRegistry,
+          IMappingConfiguration mappingConfiguration)
+      {
+        throw new NotImplementedException();
+      }
+    }
```

Comment:
```suggestion
    public class FakeStorageObjectFactory : IStorageObjectFactory
    {
      public StorageProvider CreateStorageProvider (StorageProviderDefinition storageProviderDefinition, IPersistenceExtension persistenceExtension)
      {
        throw new NotImplementedException();
      }

      public StorageProvider CreateReadOnlyStorageProvider (StorageProviderDefinition storageProviderDefinition, IPersistenceExtension persistenceExtension)
      {
        throw new NotImplementedException();
      }

      public IPersistenceModelLoader CreatePersistenceModelLoader (
          StorageProviderDefinition storageProviderDefinition, IStorageProviderDefinitionFinder storageProviderDefinitionFinder)
      {
        throw new NotImplementedException();
      }

      public IDomainObjectQueryGenerator CreateDomainObjectQueryGenerator (
          StorageProviderDefinition storageProviderDefinition,
          IMethodCallTransformerProvider methodCallTransformerProvider,
          ResultOperatorHandlerRegistry resultOperatorHandlerRegistry,
          IMappingConfiguration mappingConfiguration)
      {
        throw new NotImplementedException();
      }
    }

```

newline after, not before :)

Naming: I believe we usually add "Fake" or "Testable" as a prefix, not a suffix.

Reply:
I also renamed the derived Fakes, so that "Fake" is a prefix.

For the original RM-6996 branch, I created a commit with message "fixup f24f1d2 RM-9129 Add ReadOnlyConnectionString to IStorageObjectFactory"